src/mapping/LinearCellInterpolation.cpp

```diff
@@ -0,0 +1,118 @@
+#include "LinearCellInterpolation.hpp"
+
+#include <Eigen/Core>
+#include <algorithm>
+#include <deque>
+#include <memory>
+#include <ostream>
+#include <unordered_set>
+#include <utility>
+
+#include "logging/LogMacros.hpp"
+#include "mapping/Mapping.hpp"
+#include "mapping/Polation.hpp"
+#include "math/differences.hpp"
+#include "mesh/Data.hpp"
+#include "mesh/Mesh.hpp"
+#include "mesh/SharedPointer.hpp"
+#include "mesh/Vertex.hpp"
+#include "query/Index.hpp"
+#include "utils/Event.hpp"
+#include "utils/Statistics.hpp"
+#include "utils/assertion.hpp"
+
+namespace precice {
+extern bool syncMode;
+
+namespace mapping {
+
+VolumeCellInterpolation::VolumeCellInterpolation(
+    Constraint constraint,
+    int        dimensions)
+    : BarycentricBaseMapping(constraint, dimensions)
+{
+  if (constraint == CONSISTENT) {
+    setInputRequirement(Mapping::MeshRequirement::FULL);
+    setOutputRequirement(Mapping::MeshRequirement::VERTEX);
+  } else if (constraint == CONSERVATIVE) {
+    setInputRequirement(Mapping::MeshRequirement::VERTEX);
+    setOutputRequirement(Mapping::MeshRequirement::FULL);
+  } else {
+    PRECICE_ASSERT(constraint == SCALEDCONSISTENT, constraint);
+    setInputRequirement(Mapping::MeshRequirement::FULL);
+    setOutputRequirement(Mapping::MeshRequirement::FULL);
+  }
+
+  PRECICE_CHECK(constraint != SCALEDCONSISTENT, "Volume mapping doesn't support scaled-consistent mappings.");
+}
+
+void VolumeCellInterpolation::computeMapping()
+{
+  PRECICE_TRACE(input()->vertices().size(), output()->vertices().size());
+  const std::string     baseEvent = "map.vci.computeMapping.From" + input()->getName() + "To" + output()->getName();
+  precice::utils::Event e(baseEvent, precice::syncMode);
+  PRECICE_CHECK(getDimensions() == 2, "Volume mapping not available in 3D.");
```

Comment:
This check should be in the constructor. The earlier this triggers, the better.

Let's use a `PRECICE_ASSERT` here instead.